src/Protocol/Protocol_1_8.cpp

```diff
@@ -3918,3 +3850,46 @@ void cProtocol_1_8_0::WriteEntityProperties(cPacketizer & a_Pkt, const cEntity &
 
 	a_Pkt.WriteBEInt32(0);  // NumProperties
 }
+
+
+
+
+
+UInt8 cProtocol_1_8_0::GetProtocolEntityType(const cEntity & a_Entity)
+{
+	using Type = cEntity::eEntityType;
+
+	switch (a_Entity.GetEntityType())
+	{
+		case Type::etEnderCrystal: return 51;
+		case Type::etPickup: return 2;
+		case Type::etFallingBlock: return 70;
+		case Type::etMinecart: return 10;
+		case Type::etBoat: return 1;
+		case Type::etTNT: return 50;
+		case Type::etProjectile:
+		{
+			using PType = cProjectileEntity::eKind;
+			const auto & Projectile = static_cast<const cProjectileEntity &>(a_Entity);
+
+			switch (Projectile.GetProjectileKind())
+			{
+				case PType::pkArrow: return 60;
+				case PType::pkSnowball: return 61;
+				case PType::pkEgg: return 62;
+				case PType::pkGhastFireball: return 63;
+				case PType::pkFireCharge: return 64;
+				case PType::pkEnderPearl: return 65;
+				case PType::pkExpBottle: return 75;
+				case PType::pkSplashPotion: return 73;
+				case PType::pkFirework: return 76;
+				case PType::pkWitherSkull: return 66;
+				default: UNREACHABLE("Spawned an unhandled projectile");
+			}
```

Comment:
The idea is for the compiler to give you warnings if you don't cover all the enum cases. That only happens if there isn't a `default` label.
```suggestion
			}
			UNREACHABLE("Spawned an unhandled projectile");
```

Reply:
(This happens in a few other places as well)